src/geff/metadata_schema.py

```diff
@@ -135,6 +135,53 @@ class DisplayHint(BaseModel):
     )
 
 
+class PropMetadata(BaseModel):
+    """Metadata describing a property in the geff graph."""
+
+    identifier: str
+    dtype: str  # TODO: investigate how other packages deal with data types
+    unit: str
+    name: str | None = None
+    shortname: str | None = None  # TODO: do we keep this?
+    description: str | None = None
+
+    @model_validator(mode="after")
+    def _validate_model(self) -> PropMetadata:
+        if not self.identifier:
+            raise ValueError("Property identifier cannot be an empty string.")
+        if not self.dtype:
+            raise ValueError("Property dtype cannot be an empty string.")
```

Comment:
I think pydantic should be doing these checks for you and you don't need to have them explicitly